### PR TITLE
InvalidationFlagger continued: Try reporting in the check summary

### PR DESCRIPTION
**Pull request by IanButterworth**
Continuation of #8

### CODE REVIEW

File: .github/workflows/InvalidationFlagger.yml [hunk 1]
```diff
@@ -25,6 +25,11 @@ jobs:
         echo "Invalidations on master: ${{ steps.invs_master.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps)"
         echo "This branch: ${{ steps.invs_pr.outputs.total }} (${{ steps.invs_pr.outputs.deps }} via deps)"
       shell: bash
+    - name: Report invalidations on check conclusion
+      uses: dflydev/check-runs-action@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        output: "{title: '${{ steps.invs_pr.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps) vs. ${{ steps.invs_master.outputs.total }} on master' }"
```
Comment (johnnychen94) [inline .github/workflows/InvalidationFlagger.yml]: probably try this

```suggestion
        output: |
          title: invalidation checks
          summary: ${{ steps.invs_pr.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps) vs. ${{ steps.invs_master.outputs.total }} on master
```

File: .github/workflows/InvalidationFlagger.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
       uses: dflydev/check-runs-action@v1
       with:
         token: ${{ secrets.GITHUB_TOKEN }}
-        output: "{title: '${{ steps.invs_pr.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps) vs. ${{ steps.invs_master.outputs.total }} on master' }"
+        output: "{'title' : '${{ steps.invs_pr.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps) vs. ${{ steps.invs_master.outputs.total }} on master' }"
```
Comment (johnnychen94) [inline .github/workflows/InvalidationFlagger.yml]: let's try swapping `'` and `"` here

File: .github/workflows/InvalidationFlagger.yml [hunk 3]
```diff
@@ -1,5 +1,9 @@
 name: Invalidations
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches:
+      - master
```
Comment (johnnychen94) [inline .github/workflows/InvalidationFlagger.yml]:
```suggestion
on: [push]
```

The checks run seems only to work on a push event right now.

> https://docs.github.com/en/rest/reference/checks#check-runs
> Note: The Checks API only looks for pushes in the repository where the check suite or check run were created. Pushes to a branch in a forked repository are not detected and return an empty pull_requests array.

See https://github.com/johnnychen94/CheckTest.jl/pull/2 for a demo

File: .github/workflows/InvalidationFlagger.yml [hunk 4]
```diff
@@ -20,14 +21,18 @@ jobs:
...
-        if (( ${{ steps.invs_pr.outputs.total }} > ${{ steps.invs_master.outputs.total }} )); then
-            exit 1
-        fi
-      shell: bash
+      shell: bash  
+      
+    - name: Report invalidations to check
+      uses: dflydev/check-runs-action@v1
+      with:
+        token: ${{ secrets.IAN_PAT }}
```
Comment (johnnychen94) [inline .github/workflows/InvalidationFlagger.yml]: Need to restore this to `secrets.GITHUB_TOKEN`

Hope this is the last patch to make things work

```suggestion
        token: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by johnnychen94** [comment]
🎉

File: .github/workflows/InvalidationFlagger.yml [hunk 5]
```diff
@@ -20,14 +21,29 @@ jobs:
...
+      
+    - name: Report increased invalidations (fail)
+      if: ${{ steps.invs_pr.outputs.total > steps.invs_master.outputs.total }}
+      uses: dflydev/check-runs-action@v1
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        output: '{"summary":Invalidations on master: ${{ steps.invs_master.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps)\nThis branch: ${{ steps.invs_pr.outputs.total }} (${{ steps.invs_pr.outputs.deps }} via deps)"}'
+        id: report
+        name: "${{ steps.invs_pr.outputs.total }} (${{ steps.invs_master.outputs.deps }} via deps) vs. ${{ steps.invs_master.outputs.total }} on master"
+        conclusion: "failure"
```
Comment (johnnychen94) [inline .github/workflows/InvalidationFlagger.yml]: we could remove the `if` check and change the `conclusion` to:

```
        conclusion: ${{ steps.invs_pr.outputs.total > steps.invs_master.outputs.total }} ? "failure" : "success"
```

This reduces the code amounts a bit.

Comment (IanButterworth) [reply .github/workflows/InvalidationFlagger.yml]: Tried this and `${{ steps.invs_pr.outputs.total > steps.invs_master.outputs.total ? "failure" : "success" }}` but both failed to be parsed

Comment (johnnychen94) [reply .github/workflows/InvalidationFlagger.yml]: Oh my bad, it should be

`${{ steps.invs_pr.outputs.total }} > ${{ steps.invs_master.outputs.total }} ? "failure" : "success"`

each variable should be wrapped by `${{ ... }}` in YAML format

**Comment by IanButterworth** [comment]
I don't _love_ that this creates another check action to do the report.. but it is now functional. If there's a way to update the `Invalidations / evaluate` check then that would be optimal

btw, thanks for the help @johnnychen94 !

File: .github/workflows/InvalidationFlagger.yml (hunk 5 above)
Comment (IanButterworth) [reply .github/workflows/InvalidationFlagger.yml]: that still wouldn't parse unfortunately. I broke it up with the code duplication

**Comment by IanButterworth** [comment]
It also seems a little buggy.. sometimes the report check adopts the `Run tests` prefix, rather than the `Invalidations` prefix, as happened in that last run

**Comment by johnnychen94** [comment]
Similar to #8 I feel this should be wrapped into one action so that others don't need to tweak this. Note that the checks run is still a preview feature and it would be better to handle this instability in upstream `julia-actions/julia-invalidations`.

Composite action can't use `use` right now, I'll see if I can wrap this function up in a Julia script using `Github.jl` and work that limitation around, probably this weekend.

Feel free to merge this if you think the current version is ready.

**Pull request closed without merging**